tests/meta/transfer/test_external_data.py

```diff
@@ -60,38 +60,47 @@ def test_get_destination_path(
     assert destination_path == Path("/path/on/hasta/cust000/ADM1/")
 
 
-def test_download_sample(
-    apps_dir: Path,
-    customer_id: str,
-    cust_sample_id: str,
+def test_transfer_sample_files_from_source(
+    caplog,
+    cg_context: CGConfig,
+    customer_id,
+    cust_sample_id,
     external_data_api: ExternalDataAPI,
+    external_data_directory,
+    helpers,
     mocker,
-    sample_id: str,
-    ticket_nr: int,
+    sample_store,
+    ticket_nr,
 ):
-    """Test for transferring external data via SLURM"""
+    caplog.set_level(logging.INFO)
 
-    # GIVEN paths needed to run rsync
-    mocker.patch.object(ExternalDataAPI, "create_log_dir")
-    ExternalDataAPI.create_log_dir.return_value = apps_dir
+    # GIVEN a Store with three samples, where only two samples are present in the source folder
+    helpers.add_sample(
+        store=external_data_api.status_db, name=cust_sample_id + "1", ticket=ticket_nr
+    )
+    helpers.add_sample(
+        store=external_data_api.status_db, name=cust_sample_id + "2", ticket=ticket_nr
+    )
+    helpers.add_sample(
+        store=external_data_api.status_db, name=cust_sample_id + "3", ticket=ticket_nr
```

Comment:
Since you are reusing: cust_sample_id + "1", cust_sample_id + "2", cust_sample_id + "3" later I think you should make them into variables in this test instead so that they are defined only once and then reused.